kubexecutor/src/main/kotlin/pl/edu/prz/kod/executor/JavaExecutor.kt

```diff
@@ -0,0 +1,42 @@
+package pl.edu.prz.kod.executor
+
+import pl.edu.prz.kod.domain.ExecutionResult
+import pl.edu.prz.kod.domain.Language
+import pl.edu.prz.kod.error.CompilationFailedError
+import java.io.File
+import java.util.regex.Pattern
+
+class JavaExecutor: AbstractExecutor(Language.JAVA) {
+    private val srcDirPath = "/app/executor/src"
+    private val buildDirPath = "/app/executor/build"
+    private val runDirPath = "/app/executor/run"
+
+    override fun execute(code: String): ExecutionResult {
+        val srcDir = File(srcDirPath)
+        srcDir.mkdirs()
+        val runDir = File(runDirPath)
+        runDir.mkdirs()
+
+        val classNameMatcher = Pattern.compile("class\\s+([^\\s{]+)[\\s{]").matcher(code)
+        var javaClassName = "Main"
+        if (classNameMatcher.find()) {
+            javaClassName = classNameMatcher.group(1)
+        }
```

Comment:
How about doing it this way?
```suggestion
   val javaClassName = if (classNameMatcher.find()) classNameMatcher.group(1) else "Main"
```

Reply:
Good idea, fixed.